qemu/tests/multi_nics_verify.py

```diff
@@ -35,39 +42,66 @@ def check_nics_num(expect_c, session):
             return (False, msg)
         return (True, msg + 'Nics count match')
 
+    nics_num = int(params.get("nics_num", 8))
+    for i in range(nics_num):
+        nics = "nic%s" % i
+        params["nics"] = ' '.join([params["nics"], nics])
+    params["start_vm"] = "yes"
+    env_process.preprocess_vm(test, params, env, params["main_vm"])
+
     vm = env.get_vm(params["main_vm"])
     vm.verify_alive()
     session = vm.wait_for_login(timeout=int(params.get("login_timeout", 360)))
-    # Redirect ifconfig output from guest to log file
-    log_file = os.path.join(test.debugdir, "ifconfig")
-    ifconfig_output = session.cmd("ifconfig")
-    log_file_object = open(log_file, "w")
-    log_file_object.write(ifconfig_output)
 
     # Get the ethernet cards number from params
-    nics_num = len(params.objects("nics"))
     logging.info("[ %s ] NICs card specified in config file" % nics_num)
 
-    # Pre-judgement for the ethernet interface
-    logging.debug(check_nics_num(nics_num, session)[1])
-    txt = "Create configure file for every NIC interface in guest."
-    error_context.context(txt, logging.info)
-    ifname_list = utils_net.get_linux_ifname(session)
-    ifcfg_path = "/etc/sysconfig/network-scripts/ifcfg-%s"
-    for ifname in ifname_list:
-        eth_config_path = ifcfg_path % ifname
-        eth_config = "DEVICE=%s\\nBOOTPROTO=dhcp\\nONBOOT=yes" % ifname
-        cmd = "echo -e '%s' > %s" % (eth_config, eth_config_path)
-        s, o = session.cmd_status_output(cmd)
-        if s != 0:
-            err_msg = "Failed to create ether config file: %s\nReason is: %s"
-            test.error(err_msg % (eth_config_path, o))
+    os_type = params.get("os_type", "linux")
+    if params.get("redirect_log_file", "yes") == "no":
+        # Check all the interfaces in guest get ips
+        guest_ifname = ""
+        guest_ip = ""
+        nic_interface = []
+        for index, nic in enumerate(vm.virtnet):
+            logging.info("index %s nic" % index)
+            guest_ip = utils_net.get_guest_ip_addr(session, nic.mac, os_type,
+                                                   ip_version="ipv4")
+            if not guest_ip:
+                err_log = "vm get interface %s's ip failed." % index
+                test.fail(err_log)
+            nic_interface.append(guest_ip)
+        nic_interface_len = len(nic_interface)
+        if nics_num != nic_interface_len:
+            test.fail("Not all the nics in the guest have ips!")
+        else:
```

Comment:
No need to add `else`, `logging.info` is ok.

Reply:
i think to keep the else or not is not so rigorous.
both method are fine.